docs/how_to/create_rectangle_on_map.rst

```diff
@@ -0,0 +1,76 @@
+.. _sunpy-how-to-create-rectangle-on-map:
+
+************************************
+How to Draw a Rectangle on sunpy map
+************************************
+
+SunPy provides a convenient method called :meth:`~sunpy.map.GenericMap.draw_quadrangle` to draw rectangles on maps.
+In this guide, we'll demonstrate four different ways to draw a rectangle on a map using SunPy.
+
+Specify Corners as a Single `~astropy.coordinates.SkyCoord` Object
+==================================================================
+
+Taking two opposite corners as a single SkyCoord object
+
+.. code-block:: python
+
+    >>> import matplotlib.pyplot as plt
+
+    >>> import astropy.units as u
+    >>> from astropy.coordinates import SkyCoord
+
+    >>> import sunpy.data.sample
+    >>> import sunpy.map
+
+    >>> aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+    >>> fig = plt.figure(figsize=(5, 5))
+    >>> ax = fig.add_subplot(projection=aia_map)
+    >>> aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent) # doctest: +SKIP
+    >>> coords = SkyCoord(Tx=(100, 500) * u.arcsec, Ty=(200, 500) * u.arcsec,frame=aia_map.coordinate_frame)
+    >>> aia_map.draw_quadrangle(coords, axes=ax, edgecolor="blue") # doctest: +SKIP
+    >>> plt.show() # doctest: +SKIP
+
+Specify Corners as Separate `~astropy.coordinates.SkyCoord` Objects
+===================================================================
+
+Taking two opposite corners as separate SkyCoord Objects
+
+.. code-block:: python
+
+    >>> aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+    >>> fig = plt.figure(figsize=(5, 5))
+    >>> ax = fig.add_subplot(projection=aia_map)
+    >>> aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent) # doctest: +SKIP
+    >>> bottom_left = SkyCoord(-500 * u.arcsec, 200 * u.arcsec, frame=aia_map.coordinate_frame)
+    >>> top_right = SkyCoord(-100 * u.arcsec, 500 * u.arcsec, frame=aia_map.coordinate_frame)
+    >>> aia_map.draw_quadrangle(bottom_left, axes=ax, top_right=top_right, edgecolor="green") # doctest: +SKIP
+    >>> plt.show() # doctest: +SKIP
+
+Specify One Corner, Width, and Height
+=====================================
+
+.. code-block:: python
+
+    >>> aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+    >>> fig = plt.figure(figsize=(5, 5))
+    >>> ax = fig.add_subplot(projection=aia_map)
+    >>> aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent) # doctest: +SKIP
+    >>> bottom_left = SkyCoord(-500 * u.arcsec, -500 * u.arcsec, frame=aia_map.coordinate_frame)
+    >>> width = 400 * u.arcsec
+    >>> height = 300 * u.arcsec
+    >>> aia_map.draw_quadrangle(bottom_left, axes=ax, width=width, height=height, edgecolor="yellow") # doctest: +SKIP
+    >>> plt.show() # doctest: +SKIP
+
+Draw a Rectangle in Pixel Coordinates
+=====================================
```

Comment:
```suggestion
Using pixel coordinates
-----------------------

We will use a `~astropy.coordinates.SkyCoord` to work out the pixel coordinates instead of using coordinates as we do above.
```